autopilot/top_centrality_test.go

```diff
@@ -0,0 +1,108 @@
+package autopilot
+
+import (
+	"testing"
+
+	"github.com/btcsuite/btcutil"
+	"github.com/stretchr/testify/require"
+)
+
+// TestTopCentralityEmptyGraph tests that we don't accidentally return a score
+// for an empty graph.
+func TestTopCentralityEmptyGraph(t *testing.T) {
+	topCentrality := NewTopCentrality()
+
+	pub, err := randKey()
+	require.NoError(t, err, "unable to generate key")
+
+	nodes := map[NodeID]struct{}{
+		NewNodeID(pub): {},
+	}
+
+	for _, chanGraph := range chanGraphs {
+		graph, cleanup, err := chanGraph.genFunc()
+		success := t.Run(chanGraph.name, func(t *testing.T) {
+			require.NoError(t, err, "unable to create graph")
+			if cleanup != nil {
+				defer cleanup()
+			}
+
+			const chanSize = btcutil.SatoshiPerBitcoin
+
+			// Attempt to get centrality scores for the empty
+			// graph, and expect no scores returned.
+			scores, err := topCentrality.NodeScores(
+				graph, nil, chanSize, nodes,
+			)
+
+			require.NoError(t, err)
+			require.Equal(t, 0, len(scores))
+		})
+
+		require.True(t, success)
+	}
+}
+
+// TestTopCentralityNonEmptyGraph tests that we return the correct normalized
+// centralitiy values given a non empty graph, correctly filtered down to the
+// passed nodes and omitting nodes which we have channels with.
+func TestTopCentralityNonEmptyGraph(t *testing.T) {
```

Comment:
Could these tests be flattened into one? Since we're getting node scores and asserting length for both? The tests could just have a `buildGraph` function which provides the graph we want, rather than having two similar tests except for this one input.

Reply:
You raised a great question. Actually they were already kind of flattened as by adding the empty node set and empty channel set we can include the other test in in this one. Made some structural changes to (hopefully) make it more readable too and extended the tests to cover connectivity from none, to full.